app/src/main/java/com/example/sharewhatyoucanproject/DonorActivity.kt

```diff
@@ -0,0 +1,166 @@
+import java.io.IOException
+import java.util.ArrayList
+
+class DonorActivity : AppCompatActivity()
+{
+
+
+    // Folder path for Firebase Storage.
+    var Storage_Path = "All_Image_Uploads/"
+
+    // Root Database Name for Firebase Database.
+    var Database_Path = "All_Image_Uploads_Database"
+
+    // Creating button.
+    var ChooseButton: Button? = null
+    var UploadButton: Button? = null
+
+
+    // Creating EditText.
+    var ImageName: EditText? = null
+    var ImageDescription: EditText? = null
+
+    // Creating ImageView.
+    var SelectImage: ImageView? = null
+
+    // Creating URI.
+    var FilePathUri: Uri? = null
+
+    // Creating StorageReference and DatabaseReference object.
+    var storageReference: StorageReference? = null
+    var databaseReference: DatabaseReference? = null
+
+    // Image request code for onActivityResult() .
+    var Image_Request_Code = 7
+    var progressDialog: ProgressDialog? = null
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_donor)
+
+
+        // Assign FirebaseStorage instance to storageReference.
+        storageReference = FirebaseStorage.getInstance().reference
+
+        // Assign FirebaseDatabase instance with root database name.
+        databaseReference = FirebaseDatabase.getInstance().getReference(Database_Path)
+
+        //Assign ID'S to button.
+        ChooseButton = findViewById<View>(R.id.btnChoose) as Button
+        UploadButton = findViewById<View>(R.id.btnUpload) as Button
+
+
+        // Assign ID's to EditText.
+        ImageName = findViewById<View>(R.id.foodname) as EditText
+        ImageDescription = findViewById<View>(R.id.fooddesc) as EditText
+
+        // Assign ID'S to image view.
+        SelectImage = findViewById<View>(R.id.imageView) as ImageView
+
+        // Assigning Id to ProgressDialog.
+        progressDialog = ProgressDialog(this@DonorActivity)
+
+        // Adding click listener to Choose image button.
+        ChooseButton!!.setOnClickListener { // Creating intent.
+            val intent = Intent()
+
+            // Setting intent type as image to select image from phone storage.
+            intent.type = "image/*"
+            intent.action = Intent.ACTION_GET_CONTENT
+            startActivityForResult(
+                Intent.createChooser(intent, "Please Select Image"),
+                Image_Request_Code
+            )
+        }
+
+
+        // Adding click listener to Upload image button.
+        UploadButton!!.setOnClickListener { // Calling method to upload selected image on Firebase storage.
+            UploadImageFileToFirebaseStorage()
+        }
+    }
+
+
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (requestCode == Image_Request_Code && resultCode == RESULT_OK && data != null && data.data != null) {
```

Comment:
`requestCode == Image_Request_Code && resultCode == RESULT_OK && data != null && data.data != null` is a mouthful.  To make the code more self documenting, I would write it like this:

```
val isBlah = requestCode == Image_Request_Code && resultCode == RESULT_OK && data != null && data.data != null

if (isBlah) {
```
instead of
```
if (requestCode == Image_Request_Code && resultCode == RESULT_OK && data != null && data.data != null) {
```
Just make sure you pick a good name for `isBlah`.